src/main/java/com/adventurear/sportinderservice/model/test/TestEntity.java

```diff
@@ -0,0 +1,27 @@
+package com.adventurear.sportinderservice.model.test;
+
+import lombok.Data;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.DBRef;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import java.util.List;
+
+@Data
+@Document
+public class TestEntity {
+    @Id
+    private String id;
+    private String testString;
+    private int testInteger;
+    private boolean testBoolean;
+    @DBRef
+    private List<TestNestedEntity> testNestedEntities;
+
+    public TestEntity(String testString, int testInteger, boolean testBoolean, List<TestNestedEntity> testNestedEntities) {
```

Comment:
FYI, since we have @Data annotation, we might not need to explicitly declare a constructor. This is not a problem just a reminder that if we define a constructor we will override the one that @Data generates which has nullchecks for required fields.